google/gax/__init__.py

```diff
@@ -587,21 +594,11 @@ def add_done_callback(self, fn):  # pylint: disable=invalid-name
            operation is done or cancelled, calls the function with this
            _OperationFuture.
         """
-        def _execute_clbks(self):
-            self._poll()
-            for clbk in self._done_clbks:
-                try:
-                    clbk(self)
-                except Exception as ex:  # pylint: disable=broad-except
-                    _LOG.exception(ex)
-
-        self._done_clbks.append(fn)
+        self._queue.put(fn)
         if self._process is None:
-            self._process = multiprocessing.Process(
-                target=_execute_clbks, args=(self,))
-            self._process.start()
+            mp.Process(target=self._execute_clbks).start()
         elif not self._process.is_alive() and self._last_operation.done:
```

Comment:
I changed `add_done_callback` to use a list, so that it more closely matches the `c.f.Future` spec. I think that gets rid of the race condition as well.